t/lib/OpenQA/Test/FullstackUtils.pm

```diff
@@ -68,10 +68,19 @@ sub client_call {
     }
 }
 
+# prevents the page to reload itself (useful because elements are invalidated on reload which is hard to handle)
+sub prevent_reload {
+    my ($driver) = @_;
+
+    $driver->execute_script('window.reloadPage = function() { console.log("Page reload prevented"); };');
+}
+
 sub wait_for_result_panel {
     my ($driver, $result_panel, $desc, $fail_on_incomplete, $refresh_interval) = @_;
     $refresh_interval //= 1;
 
+    prevent_reload($driver);
+
```

Comment:
hmm, but the loop below doesn't reload - or is this magically happening then?

Reply:
Moved this now inside the loop. However, this would now reload the page very often when the test is running.

So I made my mocking of `reloadPage` set a flag (`window.wouldHaveReloaded = true;`). Only if the flag is set the manual reload is triggered (`if ($driver->execute_script('return window.wouldHaveReloaded;')) { ...`. This also allows to verify that the automatic reload is actually prevented and done manually instead (`note("reloading manually ($desc)");`).